robottelo/ssh.py

```diff
@@ -241,6 +241,22 @@ def upload_file(local_file, remote_file, key_filename=None, hostname=None):
             sftp.close()
 
 
+def upload_files(local_dir, remote_dir, file_type='.txt'):
+    """ Upload all files from directory to a remote directory
+    :param local_dir: all files from local path to be uploaded.
+    :param remote_dir: a remote path where the uploaded files will be
+           placed.
+    :param file_type: filter only files contains the type extension
+    """
+    command("mkdir -p {}".format(remote_dir))
+    for root, dirs, files in os.walk(local_dir):
+        for local_filename in files:
+            if file_type in local_filename:
+                remote_filename = "{0}{1}".format(remote_dir, local_filename)
+                local_filename = "{0}{1}".format(local_dir, local_filename)
+                upload_file(local_filename, remote_filename)
```

Comment:
I'd not reuse the `upload_file` function because then you create a new connection for every file. Reusing the SFTP connection is efficient and worth the duplication. You could introduce a helper and use it in both functions:

```python
def _upload_file(sftp, local_file, remote_file):
    # Check if local_file is a file-like object and use the proper
    # paramiko function to upload it to the remote machine.
    if hasattr(local_file, 'read'):
        sftp.putfo(local_file, remote_file)
    else:
        sftp.put(local_file, remote_file)
```

Reply:
I'll recommend sftd session name istead of sftp connection to make a diffrencr between ssh real connection and sftp session 

eg: create a context manger 
```python
@contextmanager
def get_sftp_session(hostname=None, username=None, password=None, key_filename=None, timeout=None)
    # create a new ssh connection
    with get_connection(...) as con
        # create sftp session
        try:
             sftp = conn.open_sftp()
              yield sftp
         finally:
            sftp.close()
```